### PR TITLE
[RFC] vim-patch:7.4.2051,7.4.2068,7.4.2097

### PR DESCRIPTION
**Pull request by jamessan**
#### vim-patch:7.4.2051

Problem:    No proper testing of trunc_string().
Solution:   Add a unittest for message.c.

https://github.com/vim/vim/commit/502ae4ba63561c98ac69af26cd9883bfd18d225f

#### vim-patch:7.4.2068

Problem:    Not all arguments of trunc_string() are tested.  Memory access
            error when running the message tests.
Solution:   Add another test case. (Yegappan Lakshmanan)  Make it easy to run
            unittests with valgrind.  Fix the access error.

https://github.com/vim/vim/commit/b9644433d2728e99fab874e5e33147ad95d23a31

#### vim-patch:7.4.2097

Problem:    Warning from 64 bit compiler.
Solution:   use size_t instead of int. (Mike Williams)

https://github.com/vim/vim/commit/d4f31dc45482e8db527ca044b9c3436b1e750006

### CODE REVIEW

**Comment by justinmk** [comment]
Nicely done. So none of the changes to main.c were useful?

File: test/unit/message_spec.lua [hunk 1]
```diff
@@ -4,22 +4,28 @@ local ffi = helpers.ffi
 local eq = helpers.eq
 local to_cstr = helpers.to_cstr
 
-local message = helpers.cimport('./src/nvim/message.h')
+local message = helpers.cimport('./src/nvim/message.h', './src/nvim/memory.h',
```
Comment (justinmk) [inline test/unit/message_spec.lua]: consider renaming this to `cimported` (we do this in some other tests). I think long ago cimport() didn't support multiple arguments or something, but anyways `message.xmalloc()` looks a bit weird, `cimported.xmalloc()` is a reasonable pattern.

File: test/unit/message_spec.lua [hunk 2]
```diff
@@ -0,0 +1,59 @@
+local helpers = require("test.unit.helpers")
+
+local ffi = helpers.ffi
+local eq = helpers.eq
+local to_cstr = helpers.to_cstr
+
+local message = helpers.cimport('./src/nvim/message.h', './src/nvim/memory.h',
+                                './src/nvim/strings.h')
```
Comment (justinmk) [inline test/unit/message_spec.lua]: consider renaming this to `cimported` (we do this in some other tests). I think long ago cimport() didn't support multiple arguments or something, but anyways `message.xmalloc()` looks a bit weird, `cimported.xmalloc()` is a reasonable pattern.

**Comment by jamessan** [comment]
> Nicely done. So none of the changes to main.c were useful?

The refactor in `main` to create `common_init` is essentially the same, and same motivation, that we'd already done when creating `time_init`, `init_params`, `early_init`, etc..

**Review by justinmk — APPROVED** [review]

**Pull request merged**